legacy/reconstruct_memberships.py

```diff
@@ -41,6 +41,28 @@ def interval_count(interval_list):
     return counter
 
 
+def nextmonth(original_date):
+    return (original_date.replace(day=1) + timedelta(days=32)).replace(day=1)
+
+
+def prevmonth(original_date):
+    return (original_date.replace(day=1) - timedelta(days=1)).replace(day=1)
+
+
+def advance_some_months(original_date, months):
+    current_date = original_date
+    for _ in range(months):
+        current_date = nextmonth(current_date)
+    return current_date
+
+
+def recede_some_months(original_date, months):
+    current_date = original_date
+    for _ in range(months):
+        current_date = prevmonth(current_date)
+    return current_date
+
+
```

Comment:
Because I wasn't quite sure and did it the safe way. There is no special reason, you can edit it.